app/models/ems_refresh.rb

```diff
@@ -197,6 +197,20 @@ def self.queue_merge(targets, ems, create_task = false)
       )
     end
 
+    if targets.present?
+      # If we are storing data, we want to make sure any BinaryBlob present will be tied back to this MiqQueue record
+      payload_ids = targets.map do |type, target_hash|
+        next if type != "ManagerRefresh::Target"
+        target_hash[:payload_id]
+      end.compact
+
+      if payload_ids
+        BinaryBlob
+          .where(:id => payload_ids, :resource_id => nil)
+          .update_all(:resource_id => miq_queue_record.id, :resource_type => miq_queue_record.class.base_class)
```

Comment:
can this be a race condition, where the MiqQueue row already exists and could be picked up before we write the BinaryBlob?

Reply:
I don't think so, since we limit it by primary keys of just created BinaryBlob records ```:id => payload_ids```